### ISSUE TITLE
GraphQL API Explorer unavailable after configuring remote executor

### ISSUE THREAD

**Issue by jameshbarton** [solo-io/gloo#6722]
### Gloo Edge Version

1.12.x (beta)

### Kubernetes Version

1.22.x

### Describe the bug

The GraphQL API Explorer is unavailable after configuring a remote executor schema.

### Steps to reproduce the bug

1. Clean install to a GKE cluster
```
% glooctl install gateway enterprise \
  --license-key $GLOO_KEY \
  --version $GLOO_EDGE_VERSION
```

2. Register cluster
```
% glooctl cluster register --cluster-name local --remote-context <your-k8s-context-name>
```

3. Confirm correct settings, that apiExplorer is enabled.
```
% kubectl get settings --namespace gloo-system --output yaml | grep apiExplorerEnabled
      apiExplorerEnabled: true
```

4. Confirm version
```
% glooctl version
Client: {"version":"1.12.0-beta29"}
Server: {"type":"Gateway","enterprise":true,"kubernetes":{"containers":[{"Tag":"1.12.0-beta12","Name":"discovery-ee","Registry":"quay.io/solo-io"},{"Tag":"1.12.0-beta12","Name":"extauth-ee","Registry":"quay.io/solo-io"},{"Tag":"1.12.0-beta29","Name":"gateway","Registry":"quay.io/solo-io"},{"Tag":"1.12.0-beta12","Name":"gloo-ee-envoy-wrapper","Registry":"quay.io/solo-io"},{"Tag":"1.12.0-beta12","Name":"gloo-ee","Registry":"quay.io/solo-io"},{"Tag":"1.12.0-beta12","Name":"observability-ee","Registry":"quay.io/solo-io"},{"Tag":"1.12.0-beta12","Name":"rate-limit-ee","Registry":"quay.io/solo-io"},{"Tag":"6.2.4","Name":"redis","Registry":"docker.io"}],"namespace":"gloo-system"}}
```

5. Install Upstream
```
% kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-edge-use-cases/main/graphql-remote-executor/01-countries-us.yaml
upstream.gloo.solo.io/countries-us created
```

6. Install GraphQLApi
```
% kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-edge-use-cases/main/graphql-remote-executor/02-countries-gql.yaml
graphqlapi.graphql.gloo.solo.io/countries-remote created
```

7. Install VirtualService to route to graphql endpoint
```
% kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-edge-use-cases/main/graphql-remote-executor/03-countries-vs.yaml
virtualservice.gateway.solo.io/countries-vs created
```

8. Confirm that service works as expected
```
% curl -s "$(glooctl proxy url)/graphql" -H 'Accept-Encoding: gzip, deflate, br' -H 'Content-Type: application/json' -H 'Accept: application/json' -H 'Connection: keep-alive' -H 'DNT: 1' -H 'Origin: [http://countries.trevorblades.com'](http://countries.trevorblades.com%27/) --data-binary '{"query":"{\n  country(code: \"CW\") {\n    name\n    capital\n  }\n}"}' --compressed | jq
{
  "data": {
    "country": {
      "capital": "Willemstad",
      "name": "Curacao"
    }
  }
}
```

9. Bring up dashboard
```
% glooctl dashboard
```
Navigate to APIs tab
Click through countries-remote API link

10. Observe that API Explorer is not present.
<img width="1081" alt="Screen Shot 2022-07-12 at 10 45 26 AM" src="https://user-images.githubusercontent.com/52334260/178559019-f53b2128-002f-4655-a1df-0c6eeeb1e062.png">

### Expected Behavior

The GraphQL API Explorer should work with remote executor schemas.

### Additional Context

I confirmed this with a GKE cluster running v1.22.8-gke.202

```
% glooctl version
Client: {"version":"1.12.0-beta29"}
Server: {"type":"Gateway","enterprise":true,"kubernetes":{"containers":[{"Tag":"1.12.0-beta12","Name":"discovery-ee","Registry":"quay.io/solo-io"},{"Tag":"1.12.0-beta12","Name":"extauth-ee","Registry":"quay.io/solo-io"},{"Tag":"1.12.0-beta29","Name":"gateway","Registry":"quay.io/solo-io"},{"Tag":"1.12.0-beta12","Name":"gloo-ee-envoy-wrapper","Registry":"quay.io/solo-io"},{"Tag":"1.12.0-beta12","Name":"gloo-ee","Registry":"quay.io/solo-io"},{"Tag":"1.12.0-beta12","Name":"observability-ee","Registry":"quay.io/solo-io"},{"Tag":"1.12.0-beta12","Name":"rate-limit-ee","Registry":"quay.io/solo-io"},{"Tag":"6.2.4","Name":"redis","Registry":"docker.io"}],"namespace":"gloo-system"}}
```

**Comment by Charlesthebird** [comment]
I was able to reproduce this locally, and saw that the `consoleOptions` weren't returned to the UI from the apiserver.

For the GitHub links, I had to use the raw versions: 
- https://raw.githubusercontent.com/solo-io/gloo-edge-use-cases/main/graphql-remote-executor/02-countries-gql.yaml
- virtual service link: https://raw.githubusercontent.com/solo-io/gloo-edge-use-cases/main/graphql-remote-executor/03-countries-vs.yaml

**Comment by jameshbarton** [comment]
Thanks @Charlesthebird for looking into this. Sorry about the bad github links. I edited the original ticket to fix that.

**Comment by jenshu** [comment]
this but only occurs when gloo-fed is enabled

we are missing some rbac permissions that apiserver needs in order to read the settings

**Comment by jameshbarton** [comment]
> this bug only occurs when gloo-fed is enabled

@jenshu I tested this theory as a workaround to get through my use case, but it did NOT work for me. Here's what I did:

1. Complete uninstall: `glooctl uninstall --all`

2. Reinstall with `--with-gloo-fed=false`
% glooctl install gateway enterprise \
  --with-gloo-fed=false \
  --license-key $GLOO_KEY \
  --version $GLOO_EDGE_VERSION
  
 3. Note that I still see all the `gloo-fed` pods in my cluster, so I'm not convinced that that flag is behaving as expected.
 
 4. Register my cluster: `glooctl cluster register --cluster-name local --remote-context jimb-edge-12`
 
 5. Install the upstream, graphqlapi, and VS exactly as before.
 
6. Bring up the dashboard. The API is there after cluster registration, but I see exactly the same interface as before (no GraphiQL playground).

**Comment by jenshu** [comment]
unfortunately, the `--with-gloo-fed` flag is not working, so gloo fed does indeed still get installed
(issue here: https://github.com/solo-io/gloo/issues/6022)

as a workaround you can use helm install with `--set gloo-fed.enabled=false` instead

**Comment by jenshu** [comment]
fix will be available in GlooEE v1.11.29 and v1.12.0-beta13
